src/Mod/AddonManager/AddonManager.py

```diff
@@ -161,52 +174,53 @@ def launch(self):
+        pref.SetInt("WindowWidth", self.dialog.width())
+        pref.SetInt("WindowHeight", self.dialog.height())
+        pref.SetInt("SplitterLeft", self.dialog.splitter.sizes()[0])
+        pref.SetInt("SplitterRight", self.dialog.splitter.sizes()[1])
 
         # ensure all threads are finished before closing
         oktoclose = True
-        for worker in ["update_worker","check_worker","show_worker","showmacro_worker",
-                       "macro_worker","install_worker"]:
-            if hasattr(self,worker):
-                thread = getattr(self,worker)
+        for worker in ["update_worker", "check_worker", "show_worker", "showmacro_worker",
+                       "macro_worker", "install_worker"]:
+            if hasattr(self, worker):
+                thread = getattr(self, worker)
                 if thread:
                     if not thread.isFinished():
                         oktoclose = False
 
         # all threads have finished
         if oktoclose:
-            if (hasattr(self,"install_worker") and self.install_worker) or (hasattr(self,"addon_removed") and self.addon_removed):
+            if ((hasattr(self, "install_worker") and self.install_worker) or
+                    (hasattr(self, "addon_removed") and self.addon_removed)):
                 # display restart dialog
-                from PySide import QtGui,QtCore
                 m = QtGui.QMessageBox()
-                m.setWindowTitle(translate("AddonsInstaller","Addon manager"))
+                m.setWindowTitle(translate("AddonsInstaller", "Addon manager"))
                 m.setWindowIcon(QtGui.QIcon(":/icons/AddonManager.svg"))
-                m.setText(translate("AddonsInstaller","You must restart FreeCAD for changes to take effect. Press Ok to restart FreeCAD now, or Cancel to restart later."))
+                m.setText(translate("AddonsInstaller",
+                                    "You must restart FreeCAD for changes to take "
+                                    "effect. Press Ok to restart FreeCAD now, or "
+                                    "Cancel to restart later."))
                 m.setIcon(m.Warning)
                 m.setStandardButtons(m.Ok | m.Cancel)
                 m.setDefaultButton(m.Cancel)
                 ret = m.exec_()
                 if ret == m.Ok:
-                    shutil.rmtree(self.macro_repo_dir,onerror=self.remove_readonly)
+                    shutil.rmtree(self.macro_repo_dir, onerror=self.remove_readonly)
                     # restart FreeCAD after a delay to give time to this dialog to close
-                    QtCore.QTimer.singleShot(1000,utils.restartFreeCAD)
+                    QtCore.QTimer.singleShot(1000, utils.restartFreeCAD)
             try:
-                shutil.rmtree(self.macro_repo_dir,onerror=self.remove_readonly)
-            except:
+                shutil.rmtree(self.macro_repo_dir, onerror=self.remove_readonly)
+            except Exception:
```

Comment:
We should trap specific exceptions, not just any `Exception`; this silences the code checkers but it's not ideal to handle the error (in theory we should handle only the Exceptions we are interested in, and not trap others).

Reply:
This particular exception should not be needed at all. According to the manual exceptions should trigger `onerror` callback. However, this function sometimes raises exceptions for some reason.